### PR TITLE
Mostly retail like movement + bnpcs auto attack move delay

### PR DESCRIPTION
**Pull request by XeAri**

### CODE REVIEW

File: src/world/Actor/BNpc.cpp [hunk 1]
```diff
@@ -521,7 +522,12 @@ void Sapphire::Entity::BNpc::update( int64_t currTime )
         }
 
         if( distance > minActorDistance )
-          moveTo( pHatedActor->getPos() );
+        {
+          uint64_t tick = Util::getTimeMs();
```
Comment (NotAdam) [inline src/world/Actor/BNpc.cpp]: currTime is the current time in ms, no need to get it again

File: src/world/Actor/BNpc.cpp [hunk 2]
```diff
@@ -521,7 +522,12 @@ void Sapphire::Entity::BNpc::update( int64_t currTime )
...
+          auto pTeriMgr = m_pFw->get< World::Manager::TerritoryMgr >();
+          if ( ( tick - m_lastAttack ) > 600 && pTeriMgr->isDefaultTerritory( getCurrentZone()->getTerritoryTypeId() ) )
```
Comment (NotAdam) [inline src/world/Actor/BNpc.cpp]: is there a particular reason for this check? not sure why this is needed

Comment (AriAvery) [reply src/world/Actor/BNpc.cpp]: checked retail, every bnpcs stays a few after auto attack

Comment (NotAdam) [reply src/world/Actor/BNpc.cpp]: no I mean the `pTeriMgr->isDefaultTerritory( getCurrentZone()->getTerritoryTypeId() )`

Comment (AriAvery) [reply src/world/Actor/BNpc.cpp]: I'm not sure about raid bosses or dungeons

Comment (NotAdam) [reply src/world/Actor/BNpc.cpp]: hmmm, alright, maybe leave it out for now then? not much point in doing stuff like this when we're not certain how it works

my guess is it's consistent, but would be better to know for sure before we start adding stuff like this in

Comment (AriAvery) [reply src/world/Actor/BNpc.cpp]: alright

**Pull request merged**